### PR TITLE
fix: identify account by address and network

### PR DESCRIPTION
**Pull request by janek26**

### CODE REVIEW

File: packages/extension/src/shared/wallet.service.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export const hasNewDerivationPath = (derivationPath?: string): boolean =>
   Boolean(derivationPath?.startsWith(baseDerivationPath))
 
 export const isDeprecated = ({ signer, network }: WalletAccount): boolean => {
+  console.log("isDeprecated", signer.derivationPath)
```
[the remark this answers is not in the document]
Comment (janek26) [reply packages/extension/src/shared/wallet.service.ts:12]: thanks

File: packages/extension/src/background/wallet.ts [hunk 2]
```diff
@@ -162,45 +160,45 @@ export class Wallet {
...
   private async pushAccount(account: WalletAccount) {
-    const accounts = await this.getAccounts(true)
-    const index = accounts.findIndex((a) => a.address === account.address)
-    if (index === -1) {
-      accounts.push(account)
-    } else {
-      accounts[index] = account
-    }
-    return this.store.setItem("accounts", accounts)
+    return this.setAccounts([account])
```
Comment (delaaxe) [inline packages/extension/src/background/wallet.ts:173]: why does the `pushAccount` method seem to reset all accounts except for the one pushed

File: packages/extension/src/shared/wallet.model.ts [hunk 3]
```diff
@@ -9,7 +9,14 @@ export interface WithSigner {
   signer: WalletAccountSigner
 }
 
-export interface WalletAccount extends WithSigner {
+export interface UniqAccount {
```
Comment (delaaxe) [inline packages/extension/src/shared/wallet.model.ts:12]: it's not more unique than the other interface no? anyway avoid abbreviations

```suggestion
export interface BaseWalletAccount {
```

File: packages/extension/src/ui/features/accountTokens/tokens.state.ts [hunk 4]
```diff
@@ -147,7 +147,11 @@ export const useTokensWithBalance = (): UseTokens => {
   )
 
   const { data, isValidating, error, mutate } = useSWR(
-    [selectedAccount?.address, "accountTokenBalances"],
+    [
+      selectedAccount?.address,
+      selectedAccount?.network?.id,
+      "accountTokenBalances",
+    ],
```
Comment (delaaxe) [inline packages/extension/src/ui/features/accountTokens/tokens.state.ts:154]: just for my understanding, does adding a param here reset the cache for next load?

File: packages/extension/src/ui/services/backgroundAccounts.ts [hunk 5]
```diff
@@ -24,8 +24,11 @@ export const connectAccount = ({ address, network, signer }: Account) => {
   sendMessage({ type: "CONNECT_ACCOUNT", data: { address, network, signer } })
 }
 
-export const deleteAccount = async (address: string) => {
-  sendMessage({ type: "DELETE_ACCOUNT", data: address })
+export const deleteAccount = async (address: string, networkId: string) => {
+  sendMessage({
+    type: "DELETE_ACCOUNT",
+    data: { address, network: { id: networkId } },
```
Comment (delaaxe) [inline packages/extension/src/ui/services/backgroundAccounts.ts:30]: does the networkId really need to be part of the Wallet object here, not as a separate param?

File: packages/extension/src/ui/services/backgroundAccounts.ts [hunk 6]
```diff
@@ -39,8 +42,11 @@ export const deleteAccount = async (address: string) => {
   }
 }
 
-export const hideAccount = async (address: string) => {
-  sendMessage({ type: "HIDE_ACCOUNT", data: address })
+export const hideAccount = async (address: string, networkId: string) => {
+  sendMessage({
+    type: "HIDE_ACCOUNT",
+    data: { address, network: { id: networkId } },
```
Comment (delaaxe) [inline packages/extension/src/ui/services/backgroundAccounts.ts:48]: idem

File: packages/extension/src/background/wallet.ts [hunk 7]
```diff
@@ -83,10 +83,8 @@ function calculateContractAddress(
   ])
 }
 
-export const equalAccount = (
-  a: Pick<WalletAccount, "address" | "network">,
-  b: Pick<WalletAccount, "address" | "network">,
-) => a.address === b.address && a.network.id === b.network.id
+export const equalAccount = (a: UniqAccount, b: UniqAccount) =>
```
Comment (delaaxe) [inline packages/extension/src/background/wallet.ts:86]: unrelated but this is better english

```suggestion
export const accountsEqual = (a: UniqAccount, b: UniqAccount) =>
```

File: packages/extension/src/shared/wallet.model.ts [hunk 8]
```diff
@@ -9,7 +9,14 @@ export interface WithSigner {
...
+  address: string
+  network: {
+    id: string
+  }
+}
+
+export interface WalletAccount extends UniqAccount, WithSigner {
```
Comment (delaaxe) [inline packages/extension/src/shared/wallet.model.ts:19]: is this still needed? shouldn't all `WalletAccount`s be persisted with only id?

File: packages/extension/src/ui/features/accountTokens/tokens.state.ts (hunk 4 above)
Comment (janek26) [reply packages/extension/src/ui/features/accountTokens/tokens.state.ts:154]: yes, but just for one load

File: packages/extension/src/ui/services/backgroundAccounts.ts (hunk 5 above)
Comment (janek26) [reply packages/extension/src/ui/services/backgroundAccounts.ts:30]: I just reused the `BaseWalletAccount` here
to change this we would need to transform this twice just to improve internal messaging

**Review by delaaxe — APPROVED** [review]

**Pull request merged**